addons/l10n_sa_edi/models/account_journal.py

```diff
@@ -0,0 +1,636 @@
+
+    def _l10n_sa_api_compliance_checks(self, xml_content, CCSID_data):
+        """
+            API call to the COMPLIANCE endpoint to generate a security token used for subsequent API calls
+            Requires a CSR token and a One Time Password (OTP)
+        """
+        invoice_tree = etree.fromstring(xml_content)
+
+        # Get the Invoice Hash from the XML document
+        invoice_hash_node = invoice_tree.xpath('//*[@Id="invoiceSignedData"]/*[local-name()="DigestValue"]')[0]
+        invoice_hash = invoice_hash_node.text
+
+        # Get the Invoice UUID from the XML document
+        invoice_uuid_node = invoice_tree.xpath('//*[local-name()="UUID"]')[0]
+        invoice_uuid = invoice_uuid_node.text
+
+        request_data = {
+            'body': json.dumps({
+                "invoiceHash": invoice_hash,
+                "uuid": invoice_uuid,
+                "invoice": b64encode(xml_content.encode()).decode()
+            }),
+            'header': {
+                'Authorization': self._l10n_sa_authorization_header(CCSID_data),
+                'Clearance-Status': '1'
+            }
+        }
+        return self._l10n_sa_call_api(request_data, ZATCA_API_URLS['apis']['compliance'], 'POST')
+
+    def _l10n_sa_get_api_clearance_url(self, invoice):
+        """
+            Return the API to be used for clearance. To be overridden to account for other cases, such as reporting.
+        """
+        return ZATCA_API_URLS['apis']['reporting' if invoice._l10n_sa_is_simplified() else 'clearance']
+
+    def _l10n_sa_api_clearance(self, invoice, xml_content, PCSID_data):
+        """
+            API call to the CLEARANCE/REPORTING endpoint to sign an invoice
+                - If SIMPLIFIED invoice: Reporting
+                - If STANDARD invoice: Clearance
+        """
+        invoice_tree = etree.fromstring(xml_content)
+        invoice_hash_node = invoice_tree.xpath('//*[@Id="invoiceSignedData"]/*[local-name()="DigestValue"]')[0]
+        invoice_hash = invoice_hash_node.text
+        request_data = {
+            'body': json.dumps({
+                "invoiceHash": invoice_hash,
+                "uuid": invoice.l10n_sa_uuid,
+                "invoice": b64encode(xml_content.encode()).decode()
+            }),
+            'header': {
+                'Authorization': self._l10n_sa_authorization_header(PCSID_data),
+                'Clearance-Status': '1'
+            }
+        }
+        url_string = self._l10n_sa_get_api_clearance_url(invoice)
+        return self._l10n_sa_call_api(request_data, url_string, 'POST')
+
+    # ====== Certificate Methods =======
+
+    def _l10n_sa_request_compliance_csid(self, otp):
+        """
+            Generate company Compliance CSID data
+        """
+        self.ensure_one()
+        CCSID_data = self._l10n_sa_api_get_compliance_CSID(otp)
+        if not CCSID_data.get('error'):
+            json.dumps(CCSID_data)
```

Comment:
dumps to where? why does this method exists? seems redundant with `_l10n_sa_get_compliance_CSID`